controllers/opentelemetrycollector_controller.go

```diff
@@ -127,7 +128,42 @@ func (r *OpenTelemetryCollectorReconciler) findOtelOwnedObjects(ctx context.Cont
 	for i := range pdbList.Items {
 		ownedObjects[pdbList.Items[i].GetUID()] = &pdbList.Items[i]
 	}
+	if params.Config.CreateRBACPermissions() == rbac.Available {
+		clusterObjects, err := r.findClusterRoleObjects(ctx, params)
+		if err != nil {
+			return nil, err
+		}
+		for k, v := range clusterObjects {
+			ownedObjects[k] = v
+		}
+	}
+	return ownedObjects, nil
+}
 
+// The cluster scope objects do not have owner reference.
+func (r *OpenTelemetryCollectorReconciler) findClusterRoleObjects(ctx context.Context, params manifests.Params) (map[types.UID]client.Object, error) {
```

Comment:
Given we have to do this for a fair amount of resources now, I think it may be worth it to make this a generic method that works on [T client.Object]

Reply:
This needs to be a list e.g. `[]metav1.List`` but then I am not able to get the UID for the map

Reply:
I think you may be able to `[]v1.Object` (from the [metav1 package](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Object))

Reply:
Then we cannot use the `client.List` and rely on labels like other methods in this file. Instead we would need to change to `client.Get` and pass as well the `[] types.NamespacedName`.

Reply:
I cannot make the method more generic and look simple